pipeline_tools/shared/submission/confirm_submission.py

```diff
@@ -94,17 +94,19 @@ def main():
     args = parser.parse_args()
     http_requests = HttpRequests()
     try:
-        wait_for_valid_status(args.envelope_url, http_requests)
+        response = wait_for_valid_status(args.envelope_url, http_requests)
```

Comment:
yeah, i believe this part will timeout if the submission is Invalid. Since the workflow is the owner of the submission, if it's not doing any correction to the metadata (I don't think there is). I think it's safe to fail early here if the submission goes invalid.